scripts/coveralls-branch-name.js

```diff
@@ -0,0 +1,17 @@
+/* global process require */
+
+const {execSync} = require('child_process');
+
+const nameRevCmd = 'git name-rev --name-only HEAD';
+let branchName = execSync(nameRevCmd).toString().trim();
+
+if (branchName.startsWith('remotes/pull')) {
+  execSync('git checkout "HEAD^2"', {stdio: 'ignore'});
+  const _branchName = branchName;
+  branchName = execSync(nameRevCmd).toString().trim().split('/').slice(2).join('/');
+  execSync(`git checkout ${_branchName}`, {stdio: 'ignore'});
```

Comment:
Sorry, this is pretty confusing, just trying to make sense of it. Is this saying, "if the current branch name starts with `remotes/pull`, then checkout two commits back (in detached head mode), get the current branch name, then re-checkout the original branch (to restore azp functionality presumably)?

Reply:
It has to do with the fact the coveralls doesn't officially support Azure Pipelines, which is something I don't think we realized when switching away from Travis and AppVeyor.

When a PR build (as opposed to a branch build) is running on AzP, the branch name reported by `git name-rev --name-only HEAD` will be something like `remotes/pull/1556/merge`, but that's not a branch name that can be resolved via `https://github.com/embark-framework/embark/tree/remotes/pull/1556/merge`, it will be a 404. By checking out `HEAD^2` we can get the name of branch that used as the basis for the PR, e.g. `fix/coveralls-azp`.